diff --git a/config/compilers/Makefile b/config/compilers/Makefile
index 3d3d5bb..e21bf9b 100644
--- a/config/compilers/Makefile
+++ b/config/compilers/Makefile
@@ -41,7 +41,7 @@ COMPILERS = gcc.mak gcc27.mak gcc28.mak gcc295.mak \
 	egcs.mak gcc27emx.mak intel80.mak \
         suncc.mak suncc40.mak \
         jolt.mak jdk.mak jdk12.mak none.mak \
-	jikes.mak gcc296.mak gcc30.mak gcc31.mak gcc32.mak
+	jikes.mak gcc296.mak gcc30.mak gcc31.mak gcc32.mak gcc42.mak
 
 FILES = Makefile $(COMPILERS) gcc_defaults.mak suncc_defaults.mak jdk_defaults.mak jikes_defaults.mak
 
diff --git a/config/compilers/gcc42.mak b/config/compilers/gcc42.mak
new file mode 100644
index 0000000..3bbf897
--- /dev/null
+++ b/config/compilers/gcc42.mak
@@ -0,0 +1,62 @@
+ ###########################################################################
+ ##                                                                       ##
+ ##                Centre for Speech Technology Research                  ##
+ ##                     University of Edinburgh, UK                       ##
+ ##                         Copyright (c) 1999                            ##
+ ##                        All Rights Reserved.                           ##
+ ##                                                                       ##
+ ##  Permission is hereby granted, free of charge, to use and distribute  ##
+ ##  this software and its documentation without restriction, including   ##
+ ##  without limitation the rights to use, copy, modify, merge, publish,  ##
+ ##  distribute, sublicense, and/or sell copies of this work, and to      ##
+ ##  permit persons to whom this work is furnished to do so, subject to   ##
+ ##  the following conditions:                                            ##
+ ##   1. The code must retain the above copyright notice, this list of    ##
+ ##      conditions and the following disclaimer.                         ##
+ ##   2. Any modifications must be clearly marked as such.                ##
+ ##   3. Original authors' names are not deleted.                         ##
+ ##   4. The authors' names are not used to endorse or promote products   ##
+ ##      derived from this software without specific prior written        ##
+ ##      permission.                                                      ##
+ ##                                                                       ##
+ ##  THE UNIVERSITY OF EDINBURGH AND THE CONTRIBUTORS TO THIS WORK        ##
+ ##  DISCLAIM ALL WARRANTIES WITH REGARD TO THIS SOFTWARE, INCLUDING      ##
+ ##  ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS, IN NO EVENT   ##
+ ##  SHALL THE UNIVERSITY OF EDINBURGH NOR THE CONTRIBUTORS BE LIABLE     ##
+ ##  FOR ANY SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES    ##
+ ##  WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN   ##
+ ##  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,          ##
+ ##  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF       ##
+ ##  THIS SOFTWARE.                                                       ##
+ ##                                                                       ##
+ ###########################################################################
+ ##                                                                       ##
+ ##                 Author: William Hubbs (williamh@gentoo.org)                 ##
+ ##                   Date: Sun Aug 25 2002                               ##
+ ## --------------------------------------------------------------------  ##
+ ## Settings for GCC-4.2                                                  ##
+ ##                                                                       ##
+ ###########################################################################
+
+include $(EST)/config/compilers/gcc_defaults.mak
+
+ifndef GCC42
+    GCC42 = gcc
+endif
+
+CC= $(GCC42)
+CXX = g++
+
+COMPILER_DESC=gcc-4.[234]
+
+SHARED_CCFLAGS  = -fPIC
+SHARED_CXXFLAGS  = -fPIC
+SHARED_LINKFLAGS =
+
+ ifndef GCC_MAKE_SHARED_LIB
+    MAKE_SHARED_LIB = $(CXX) -shared -o XXX -Wl,-soname,YYY
+ else
+     MAKE_SHARED_LIB = $(GCC_MAKE_SHARED_LIB)
+ endif
+
+STATIC_LINKFLAGS = -Dstatic
